pkg/remotecluster/remotecluster.go

```diff
@@ -0,0 +1,223 @@
+package remotecluster
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	v1net "k8s.io/api/networking/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+)
+
+const defaultResyncTime = time.Hour * 24
+
+type RemoteCluster struct {
+	stopCh chan struct{}
+
+	ClusterID             string
+	ClientSet             kubernetes.Interface
+	podInformer           cache.SharedIndexInformer
+	networkPolicyInformer cache.SharedIndexInformer
+
+	eventChanMutex *sync.Mutex
+	eventChan      chan *Event
+}
+
+type EventType string
+
+const (
+	AddEvent    EventType = "Added"
+	DeleteEvent EventType = "Deleted"
+	UpdateEvent EventType = "Updated"
+)
+
+type ObjectType string
+
+const (
+	NetworkPolicy ObjectType = "np"
+	Pod           ObjectType = "pod"
+)
+
+type Event struct {
+	Cluster *RemoteCluster
+	Type    EventType
+	ObjType ObjectType
+	Objs    []interface{}
+	ObjID   string
```

Comment:
`ObjType` and `ObjID` are only used by UTs for verification so I don't think they belong here.  Also, FWICT, they really shouldn't be needed as users can derive `ObjType` from the target object's type and the tests can directly verify the fields from which `ObjID` is derived.

Reply:
Yes, they are used in later patches.

I found myself calculating ObjID many times everywhere, so I moved it to the event itself.
Later down the road, on event processing I need the object ID many times, and to calculate I need to extract the objects, and fields, etc...